airbyte-cdk/python/airbyte_cdk/sources/streams/utils/stream_helper.py

```diff
@@ -34,6 +35,11 @@ def get_first_record_for_slice(stream: Stream, stream_slice: Optional[Mapping[st
     :raises StopIteration: if there is no first record to return (the read_records generator is empty)
     :return: StreamData containing the first record in the slice
     """
+    if isinstance(stream, DeclarativeStream):
+        stream.retriever.requester.exit_on_rate_limit = True  # type: ignore[attr-defined]
+    else:
+        stream.exit_on_rate_limit = True  # type: ignore[attr-defined]
```

Comment:
Should we set back the value to what it previously was after generating the first stream slice? What about the parent streams?

Reply:
updated to copy stream before setting `exit_on_rate_limit`